muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/evaluation/dialogs/delete.tsx

```diff
@@ -0,0 +1,228 @@
+  isOpen?: boolean;
+  onClose?: () => any;
+  evaluations: EvaluationState;
+  eventData: EvaluationEvent;
+  removeWorkspaceEventFromServer: RemoveWorkspaceEvent;
+}
+
+/**
+ * DeleteDialogState
+ */
+interface DeleteDialogState {}
+
+/**
+ * DeleteDialog
+ */
+class DeleteDialog extends React.Component<
+  DeleteDialogProps,
+  DeleteDialogState
+> {
+  /**
+   * constructor
+   * @param props
+   */
+  constructor(props: DeleteDialogProps) {
+    super(props);
+
+    this.handleDeleteEventClick = this.handleDeleteEventClick.bind(this);
+  }
+
+  /**
+   * handleDeleteEventClick
+   */
+  handleDeleteEventClick(closeDialog: () => any) {
+    const { eventData } = this.props;
+
+    this.props.removeWorkspaceEventFromServer({
+      identifier: eventData.identifier,
+      eventType: eventData.type,
+      onSuccess: () => {
+        const draftId = eventData.identifier;
+
+        cleanWorkspaceAndSupplementationDrafts(draftId);
+
+        closeDialog();
+      },
+      onFail: () => closeDialog(),
+    });
+  }
+
+  /**
+   * Component render method
+   * @returns JSX.Element
+   */
+  render() {
+    const { evaluationSelectedAssessmentId } = this.props.evaluations;
+
+    const studentNameString = `${evaluationSelectedAssessmentId.lastName}, ${evaluationSelectedAssessmentId.firstName}`;
+
+    let footer = (closeDialog: () => any) => {
+      return (
+        <div className="dialog__button-set">
+          <Button
+            buttonModifiers={["fatal", "standard-ok"]}
+            onClick={this.handleDeleteEventClick.bind(this, closeDialog)}
+          >
+            {this.props.i18n.text.get(
+              "plugin.evaluation.workspaceEvaluationDialog.removeDialog.removeButton",
+              studentNameString
+            )}
+          </Button>
+          <Button
+            buttonModifiers={["cancel", "standard-cancel"]}
+            onClick={closeDialog}
+          >
+            {this.props.i18n.text.get(
+              "plugin.evaluation.workspaceEvaluationDialog.removeDialog.cancelButton"
+            )}
+          </Button>
+        </div>
+      );
+    };
+    let content = (closeDialog: () => any) => {
+      return (
+        <div>
+          {this.props.i18n.text.get(
+            "plugin.evaluation.workspaceEvaluationDialog.removeDialog.description",
+            studentNameString
+          )}
+        </div>
+      );
+    };
+    return (
+      <Dialog
+        isOpen={this.props.isOpen}
+        onClose={this.props.onClose}
+        modifier="evaluation-remove-assessment"
+        title={this.props.i18n.text.get(
+          "plugin.evaluation.workspaceEvaluationDialog.removeDialog.title"
+        )}
+        content={content}
+        footer={footer}
+      >
+        {this.props.children}
+      </Dialog>
+    );
+  }
+}
+
+/**
+ * cleanWorkspaceAndSupplementationDrafts
+ * @param draftId
+ */
+export const cleanWorkspaceAndSupplementationDrafts = (draftId: string) => {
+  if (
```

Comment:
You are retrieving from local storage many times the same value, try to store the in a const; local storage is fast nevertheless, but that's just convention, makes for shorter code and less need for fetching from it all the time.